patterns/Behavioral/Template/Javascript/template.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * The concept was referred from https://refactoring.guru/
+design-patterns/template-method
+ * This example simply demonstrates how to implement the
+ * Template Design Pattern in a scenario: imagine you are
+ * standing in front of a kiosk machine at McDonald's to
+ * register, and you will receive a card in return.
+ */
+
+class CommunicateWithDevice {
```

Comment:
just curious if you could call this `API` or `System`? I understand why you've written it out, I just like short readable class names 😆 especially because [java has given me ptsd](https://www.reddit.com/media?url=https%3A%2F%2Fi.redd.it%2Fga3oggc0m2ba1.jpg)

Reply:
haha java, I feel you.
I don't think we can classify it as either an API or a system; those are different concepts. It's the process of communicating with a device